pandas/core/reshape/tile.py

```diff
@@ -383,10 +397,26 @@ def _bins_to_cuts(
             bins = unique_bins
 
     side = "left" if right else "right"
-    ids = ensure_int64(bins.searchsorted(x, side=side))
+    is_nullable_integer = is_extension_array_dtype(x.dtype) and is_integer_dtype(
+        x.dtype
+    )
+
+    if is_nullable_integer:
+        # TODO: Support other extension types somehow. We don't currently
```

Comment:
This is a bit interesting. We need to get integers for searchsorted and it doesn't really matter how the NA values are encoded since we mask them out later on.

I don't think we have anything in the interface like this right now. The closest is `factorize`. But that has specific restrictions on

1. The array being an enumeration from 0, 1, ... number of uniques
2. NA being -1.

Which is more work than we need here. Worth thinking about for the future.